prelude/toolchains/cxx.bzl

```
@@ -170,49 +203,54 @@ def _system_cxx_toolchain_impl(ctx: AnalysisContext):
         CxxPlatformInfo(name = "x86_64"),
     ]
 
-def _windows_linker_wrapper(ctx: AnalysisContext, linker: cmd_args) -> cmd_args:
-    # Linkers pretty much all support @file.txt argument syntax to insert
-    # arguments from the given text file, usually formatted one argument per
-    # line.
-    #
-    # - GNU ld: https://gcc.gnu.org/onlinedocs/gcc/Overall-Options.html
-    # - lld is command line compatible with GNU ld
-    # - MSVC link.exe: https://learn.microsoft.com/en-us/cpp/build/reference/linking?view=msvc-170#link-command-files
-    #
-    # However, there is inconsistency in whether they support nesting of @file
-    # arguments inside of another @file.
-    #
-    # We wrap the linker to flatten @file arguments down to 1 level of nesting.
-    return cmd_script(
-        ctx = ctx,
-        name = "windows_linker",
-        cmd = cmd_args(
-            ctx.attrs.linker_wrapper[RunInfo],
-            linker,
-        ),
-        os = ScriptOs("windows"),
-    )
+def _run_info(args):
+    return None if args == None else RunInfo(args = [args])
 
 system_cxx_toolchain = rule(
     impl = _system_cxx_toolchain_impl,
     attrs = {
+        "cxx_tools_info": attrs.default_only(attrs.exec_dep(providers = [CxxToolsInfo], default = select({
```

```suggestion
        "_cxx_tools_info": attrs.default_only(attrs.exec_dep(providers = [CxxToolsInfo], default = select({
```

Just to make it a little more obvious to users that this is an implementation detail of the rule (ideally we'd support `hide = True` for the docs but that's a problem for another day)